sub-packages/bionemo-scdl/src/bionemo/scdl/index/row_feature_index.py

```diff
@@ -113,26 +120,33 @@ def lookup(self, row: int, select_features: Optional[List[str]] = None) -> Tuple
         d_id = sum(mask) - 1
 
         # Retrieve the features for the identified value.
-        features = self._feature_arr[d_id]
+        features_dict = self._feature_arr[d_id]
 
         # If specific features are to be selected, filter the features.
         if select_features is not None:
-            features = features[select_features]
+            features = []
+            for feature in select_features:
+                if feature not in features_dict:
+                    raise ValueError(f"Provided feature column {feature} in select_features not present in dataset.")
+                features.append(features_dict[feature])
+        else:
+            features = [features_dict[f] for f in features_dict]
 
         # Return the features for the identified range.
         return features, self._labels[d_id]
 
     def number_vars_at_row(self, row: int) -> int:
-        """Return number of variables (legnth of the dataframe) in a given row.
+        """Return number of variables in a given row.
 
         Args:
             row (int): The row in the feature index.
 
         Returns:
             The length of the features at the row
         """
-        feats, _ = self.lookup(row=row)
-        return len(feats)
+        mask = ~(self._cumulative_sum_index > row)
```

Comment:
This is similar logic to what we have in lookup. Factor it out into another function?